### PR TITLE
enum is defined check and fallback values

### LINKED ISSUE #488: Enum mapping fallback value
Support a new property `FallbackValue` on the `MapEnumAttribute` to support unknown / unmatched enum values.
This should be supported for any mappings where the target is an enum.
Rel. https://github.com/riok/mapperly/issues/482#issuecomment-1584270386

### PR DESCRIPTION
**Pull request by latonz**
# enum is defined check and fallback values

Fixes #488

## Checklist

- [x] The existing code style is followed
- [x] The commit message follows our guidelines
- [x] Performed a self-review of my code
- [x] Hard-to-understand areas of my code are commented
- [x] The documentation is updated (as applicable)
- [x] Unit tests are added/updated
- [x] Integration tests are added/updated (as applicable, especially if feature/bug depends on roslyn or framework version in use)

### CODE REVIEW

File: src/Riok.Mapperly/AnalyzerReleases.Shipped.md [hunk 1]
```diff
@@ -100,3 +100,5 @@ RMG038  | Mapper   | Info     | An enum member could not be found on the target
 RMG039  | Mapper   | Error    | Enum source value is specified multiple times, a source enum value may only be specified once
 RMG040  | Mapper   | Error    | A target enum member value does not match the target enum type
 RMG041  | Mapper   | Error    | A source enum member value does not match the source enum type
+RMG042  | Mapper   | Error | The type of the enum fallback value does not match the target enum type
```
Comment (ni507) [inline src/Riok.Mapperly/AnalyzerReleases.Shipped.md:103]: Wrong indents for the new lines

File: src/Riok.Mapperly/Descriptors/MappingBuilders/StringToEnumMappingBuilder.cs [hunk 2]
```diff
@@ -22,23 +24,53 @@ public static class StringToEnumMappingBuilder
             .Any(x => x.IsGenericMethod);
 
         if (ctx.IsExpression)
+        {
```
Comment (ni507) [inline src/Riok.Mapperly/Descriptors/MappingBuilders/StringToEnumMappingBuilder.cs:27]: Are this brackets necessary? If yes, then also add below on L54.

Comment (latonz) [reply src/Riok.Mapperly/Descriptors/MappingBuilders/StringToEnumMappingBuilder.cs:27]: Fixed.

File: src/Riok.Mapperly/AnalyzerReleases.Shipped.md (hunk 1 above)
Comment (latonz) [reply src/Riok.Mapperly/AnalyzerReleases.Shipped.md]: Fixed.

**Review by ni507 — APPROVED** [review]

**Pull request merged**